server/middlewares/validators/shared/videos.ts

```diff
@@ -166,6 +176,55 @@ async function checkCanSeeAuthVideo (req: Request, res: Response, video: MVideoI
   return fail()
 }
 
+async function checkCanSeePasswordProtectedVideo (options: {
+  req: Request
+  res: Response
+  video: MVideo
+}) {
+  const { req, res, video } = options
+
+  const videoWithAccount = (video as MVideoAccountLight).VideoChannel?.Account?.userId
```

Comment:
Can you extract this in a function since we do the same thing in `checkCanSeeAuthVideo`?